cineast-api/src/main/java/org/vitrivr/cineast/api/websocket/handlers/queries/AbstractQueryMessageHandler.java

```diff
@@ -246,7 +248,31 @@ public final void handle(Session session, T message) {
     return objectIds;
   }
 
-  protected List<Thread> submitMetadata(Session session, String queryId, List<String> segmentIds, List<String> objectIds, Collection<String> segmentIdsForWhichMetadataIsFetched, Collection<String> objectIdsForWhichMetadataIsFetched) {
+  /**
+   * Submits all the data (e.g. {@link MediaObjectDescriptor}, {@link MediaSegmentDescriptor}) from Ids to the UI. Should be executed before sending results.
+   */
+  void submitSegmentAndObjectInformationFromIds(Session session, String queryId, List<String> segmentIds, List<String> objectIds) {
```

Comment:
Is there a reason you're not using the existing `submitSegmentAndObjectInformation` method?

Reply:
As I wanted to avoid sending duplicate MediaObjectDescriptor information I first had to retrieve the Segments and their respective objects, then apply the filtering to avoid sending already sent items and then finally send the information and metadata to the UI.